test/unittests/stack_test.cpp

```diff
@@ -67,15 +67,21 @@ TEST(stack, shrink)
     stack.push('b');
     stack.push('c');
     stack.push('d');
-    EXPECT_EQ(stack.top(), 'd');
+    {
```

Comment:
What is going on?

Reply:
Tried to workaround the false-positive bug of the newer gcc. This change will be removed now, as going back to older gcc removes the false report.